runtime/vm/CRIUHelpers.cpp

```diff
@@ -365,6 +382,16 @@ runInternalJVMCheckpointHooks(J9VMThread *currentThread)
 	/* Iterate heap objects to prepare internal hooks at checkpoint. */
 	vm->memoryManagerFunctions->j9mm_iterate_all_objects(vm, vm->portLibrary, 0, objectIteratorCallback, currentThread);
 
+	J9ClassWalkState j9ClassWalkState = {0};
+	J9Class *clazz = vm->internalVMFunctions->allClassesStartDo(&j9ClassWalkState, vm, NULL);
+	while (NULL != clazz) {
+		reloadTraceMethodSettings(currentThread, clazz);
```

Comment:
in here I think we need something like:

```
// Passes the class to all the class restore hooks
triggerClassRestoreHooks(currenthread, clazz);
```

Which means we need something like:

```
// currentThread vmThread token
// isRestore If FALSE, run the hook at checkpoint, otherwise run at restore
// hookFunc The hook function to be invoked for the internal hook record of the form "typedef BOOLEAN (*hookFunc)(struct J9VMThread *currentThread, J9Class *clazz)" where 
registerClassHook(J9VMThread *currentThread, BOOLEAN isRestore,  hookFunc hookFunc)
```

Reply:
Created a dedicated hook record for class iteration after restore, `pool_numElements()` will determine if the class iteration is required.